### ISSUE TITLE
Retrieve token

### ISSUE THREAD

**Issue by sqlninja** [hayfever/sorcery-jwt#10]
If I already have a user, say from an oauth login, is it possible to retrieve the token for that user?

I tried `@token = login_and_issue_token(@user.email, @user.crypted_password)`
but that just returns nil.

**Comment by hayfever** [comment]
It seems like in the code you gave you're using the encrypted password, I actually haven't used sorcery in a while but I believe you do `@token = login_and_issue_token(@user.email, @user.password)` for the unencrypted password and you should get a token back.

**Comment by sqlninja** [comment]
Yeah under a normal login flow that is accurate, we would receive the email/pw from the user. But with oAuth the user has logged on using a 3rd party (Twitter in my case), to which the twitter oauth callback just passes me the email (and other twitter related data) on the authenticated user.
I then use that email to look up if we have a user in our system matching what twitter sent us.

When the user is found I know have the activerecord instance of this user, but I need to pass back a response that includes the JWT token, so the apps/web can make subsequent authenticated calls.

This may not be possible with this gem, since it seems to be only generating a token based on a payload that includes both the email & pw...

Also, it appears there are discussions going on in Sorcery about including JWT tokenization natively, so this may be a moot point and not worth the effort here. Just though I'd asking if there was a way to get the token on a "known" user...

**Comment by joshbuker** [comment]
JWT support will indeed be added to Sorcery v1, see Sorcery/sorcery-rework#9 for further discussion.

**Comment by hayfever** [comment]
For a user you've already authenticated (e.g. via OAuth), don't use login_and_issue_token — that verifies credentials from scratch. Just issue the token directly:

```ruby
token = User.issue_token(id: @user.id, email: @user.email)
```

Documented in the README under "Token issuance API" (v1.0.0). Closing — happy to reopen if that doesn't cover your case.
